src/pkg/packager/actions.go

```diff
@@ -160,9 +160,12 @@ retryCmd:
 
 	select {
 	case <-timeout:
-		// If we reached this point, the timeout was reached.
-		return fmt.Errorf("command \"%s\" timed out after %d seconds", cmdEscaped, cfg.MaxTotalSeconds)
-
+		// If we reached this point, the timeout was reached or command failed with no retries.
+		if cfg.MaxTotalSeconds < 1 && cfg.MaxRetries < 1 {
+			return fmt.Errorf("command \"%s\" failed (no retries)", cmdEscaped)
```

Comment:
```suggestion
		if cfg.MaxTotalSeconds < 1 {
			return fmt.Errorf("command %q failed after %d retries", cmdEscaped, cfg.MaxRetries)
```

Reply:
Realized the logic I put in the issue was wrong - you can fall into this if you retried without a timeout being set...

Reply:
Made the suggested change, but just want to confirm with you - in that situation would it fall-through to the default branch of the select?

Reply:
Yes because the timeout would be 0 and the timeout channel would still be reached.

Current code with a maxTotalSeconds and maxRetries > 0:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/93c62291-1581-4275-86fa-564916577924)

Current code with a maxRetries > 0 but a maxTotalSeconds = 0:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/43e268d2-c1b7-4f78-a04c-a03de0c4d81b)

Your code with a maxRetries > 0 but a maxTotalSeconds = 0:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/78bdc9e4-2b29-4436-beae-3e0e7cf2a0e1)